diff --git a/facets.module b/facets.module
index e5cd82c..fcb6d69 100644
--- a/facets.module
+++ b/facets.module
@@ -54,6 +54,8 @@ function facets_theme($existing, $type, $theme, $path) {
         'show_count' => FALSE,
         'count' => NULL,
         'is_active' => FALSE,
+        'show_label' => FALSE,
+        'label' => '',
       ],
     ],
     'facets_item_list' => [
diff --git a/modules/facets_summary/src/FacetsSummaryManager/DefaultFacetsSummaryManager.php b/modules/facets_summary/src/FacetsSummaryManager/DefaultFacetsSummaryManager.php
index d0570e5..a9ed0bb 100644
--- a/modules/facets_summary/src/FacetsSummaryManager/DefaultFacetsSummaryManager.php
+++ b/modules/facets_summary/src/FacetsSummaryManager/DefaultFacetsSummaryManager.php
@@ -155,6 +155,8 @@ class DefaultFacetsSummaryManager {
         $item = [
           '#theme' => 'facets_result_item__summary',
           '#value' => $result->getDisplayValue(),
+          '#show_label' => TRUE,
+          '#label' => $result->getFacet()->label(),
           '#show_count' => $show_count,
           '#count' => $result->getCount(),
           '#is_active' => TRUE,
diff --git a/templates/facets-result-item.html.twig b/templates/facets-result-item.html.twig
index 1fb7d3d..941ec98 100644
--- a/templates/facets-result-item.html.twig
+++ b/templates/facets-result-item.html.twig
@@ -19,6 +19,9 @@
 {% if is_active %}
   <span class="facet-item__status js-facet-deactivate">(-)</span>
 {% endif %}
+{% if show_label %}
+<span class="facet-item__label">{{ label }}</span>
+{% endif %}
 <span class="facet-item__value">{{ value }}</span>
 {% if show_count %}
   <span class="facet-item__count">({{ count }})</span>
