diff --git a/src/IslandoraUtils.php b/src/IslandoraUtils.php
index a2df7589..954f768d 100644
--- a/src/IslandoraUtils.php
+++ b/src/IslandoraUtils.php
@@ -246,24 +246,19 @@ class IslandoraUtils {
     // Add field_external_uri.
     $fields[] = self::EXTERNAL_URI_FIELD;
 
-    $query = $this->entityTypeManager->getStorage('taxonomy_term')->getQuery();
-
-    $orGroup = $query->orConditionGroup();
+    $storage = $this->entityTypeManager->getStorage('taxonomy_term');
     foreach ($fields as $field) {
-      $orGroup->condition("$field.uri", $uri);
-    }
-
-    $results = $query
-      ->accessCheck(TRUE)
-      ->condition($orGroup)
-      ->execute();
-
-    if (empty($results)) {
-      return NULL;
+      $query = $storage->getQuery();
+      $results = $query
+        ->accessCheck(TRUE)
+        ->condition("$field.uri", $uri)
+        ->execute();
+      if (!empty($results)) {
+        return $storage->load(reset($results));
+      }
     }
 
-    return $this->entityTypeManager->getStorage('taxonomy_term')
-      ->load(reset($results));
+    return NULL;
   }
 
   /**
